assignment2/code/app.py

```diff
@@ -0,0 +1,88 @@
+import os
+from datetime import date, timedelta
+
+from flask import Flask, jsonify
+from flask_jwt import JWT, jwt_required
+from flask_restful import Api
+
+from db import db
+from models.sales_model import SalesModel
+from resources.customer_resource import CustomerSignIn
+from resources.product_resources import ProductList
+from resources.sales_resource import PurchaseList, Sales
+from security import authenticate, identity
+
+app=Flask(__name__)
+app.secret_key= '1234'
+api=Api(app)
+
+@app.before_first_request
+def create_tables():
+    db.create_all()
+
+
+app.config['JWT_EXPIRATION_DELTA'] = timedelta(seconds=1800)
+app.config['SQLALCHEMY_TRACK_MODIFICATIONS']= False
+app.config['SQLALCHEMY_DATABASE_URI'] =  os.environ.get('DATABASE_URI', 'sqlite:///data.db')
+
+
+# api endpoint for customer signing in the site for the first time
+api.add_resource(CustomerSignIn,'/sign_in')
+
+
+# customer login and check for valid customer
+app.config['JWT_AUTH_URL_RULE'] = '/log_in'
+jwt=JWT(app, authenticate, identity)
+
+
+# api for product purchase
+api.add_resource(Sales,'/purchase')
+
+
+# api endpoint for total sales
+@app.route('/totalsales')
+@jwt_required()
+def total_sales():
+    return jsonify({"message": f"Total number of sales for the day is - {len(SalesModel.find_by_date(date.today()).filter(SalesModel.sale_amount != 0).all())}."})
+
+
+# api end point for unique visitors
+@app.route('/uniquevisitors')
+@jwt_required()
+def unique_visitors():
+    visitors = SalesModel.find_by_date(date.today()).all()
+    unique_visitor = {}
+    for object in visitors:
```

Comment:
i am not able to understand this login. if the key is present you are adding one. otherwise you are adding to the key. Instead you can create a set of userids and find their length